### ISSUE TITLE
[ACTIVE] SWARM-1300 — readiness material-authorization authority/ref binding diagnostic

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1300]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1300
STATUS: ACTIVE_PENDING_POSTCLAIM_ARBITRATION
SWARM_LANE_KEY: DIAG|R01|PR1256|MATERIAL-AUTHORITY-DECISION-REF-BINDING|EXACTHEAD-AB60BD80|V1
PREFERRED_LANE: #12
PREFERRED_WORK_KIND: OPEN_SOURCE_REUSE_RESEARCH
SELECTED_WORK_KIND: READ_ONLY_HIGH_SEVERITY_DEFECT_DISCOVERY
PARENT_ISSUES: #15 #548 #723 #1093 #1251 #1256
TARGET_PR: #1256
TARGET_HEAD: ab60bd80689a91522c00c95dd26009debe166251
PRODUCT_OWNER: #1251
INTEGRATION_OWNER: #1238
CLAIMED_AT_UTC: GitHub issue updated_at for this reopening is authority
LEASE: six hours from this reopening updated_at

PRIOR PACKAGE / RELEASED:
- SWARM-1300 previously completed and released `AUDIT-B|D02|PR802|TRAIN-MICROBATCH-COMMIT-METRIC-ACCOUNTING|EXACTHEAD-82C43005|V1` as bounded PASS, with no Product mutation.

WHY ROTATE HERE:
- central D03/D04/D05/D06/D10 Product gates are live-owned; #1238 owns merge intake; #1251 owns readiness repair;
- #1256 is immediately before optimizer-step authorization and is already being repaired for READINESS1251-001 plus deterministic red CI;
- catching a distinct fail-open before the next repair push removes a repair→CI→fresh-audit cycle without colliding with Product ownership;
- preferred OPEN_SOURCE_REUSE_RESEARCH is farther from the first lawful optimizer step than this concrete launch-authority diagnostic.

SUSPECTED DISTINCT HIGH FINDING:
- `assess_learned20m_readiness()` authenticates `compute_authorization.decision_ref` and `training_authorization.decision_ref` only by membership in `verified_authorization_refs`;
- the adjacent `compute_authorization.authority` / `training_authorization.authority` objects are checked only by structural `_require_authority()` and are not cryptographically/semantically bound to the verified decision ref or to a role-bound independently trusted authority token;
- therefore, even after READINESS1251-001 authenticates the trusted-bindings bundle root, a candidate may plausibly preserve the externally frozen verified ref strings while substituting a different structurally valid repository/git/evidence/terminal authority object, yet still reach `material_training_authorized=true`;
- current positive test `test_explicit_material_authority_must_cover_estimated_maximum` reaches `material_training_authorized` using only `{COMPUTE_REF, TRAINING_REF}` plus generic `_authority(workflow=False)` objects, demonstrating the missing binding surface.

SECONDARY STRICT-TYPE NOTE TO CHECK, NOT THE PRIMARY CLAIM:
- `_validate_envelope()` uses Python equality for `schema_version` and `MODEL341_AUTHORITY`, so `schema_version=True == 1` and integer-valued float aliases may be accepted; report only if it survives exact analysis and is not already covered elsewhere.

OWNED SURFACE:
- read-only exact-head diagnosis of material compute/training authorization binding on PR #1256;
- no Product edits, no branch, no PR, no audit PASS, no merge;
- produce exact exploit reasoning / minimal regression and hand off to #1251 if confirmed; otherwise close NO_NEW_BLOCKER.

NON_OVERLAP:
- do not duplicate READINESS1251-001 trusted-bindings-root finding from #1253;
- do not duplicate #1257 portable-run red-CI diagnosis;
- no D03/D04/D05/D06/D10/recipe/pilot/resource mutation;
- no training launch, optimizer update, tokenizer fit, corpus/final-test access, paid compute, foreign pretrained weights, external LLM/API data intelligence.

ACCEPTANCE:
1. direct paginated current OPEN arbitration finds no earlier live exact/semantic diagnostic owner;
2. demonstrate or falsify authority-object substitution under fixed externally verified decision refs;
3. if confirmed, minimal same-lineage repair binds each material authorization decision ref to the exact authority object/identity independently, not merely to a free string;
4. adversarial regression must hold verified refs fixed, substitute authority git/evidence identity, and fail before material training authorization;
5. preserve distinct compute vs training authorization and all zero-authority checked-in state;
6. Product owner #1251 receives the finding before its next qualifying head if still applicable.

TRUTH_BOUNDARY:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_ACCESSED=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

**Comment by Oleksii-debug** [comment]
AUDIT VERDICT: PASS_BOUNDED / NO_NEW_BLOCKER

Target: PR #802 exact head `82c43005bb5db153482ae5b20a31d59240faaebc`
Scope: D02 `Trainer.train_microbatch` commit truth + StepMetrics/token/loss accounting only.

Independent findings:
- `optimizer_stepped=True` is returned only on an accumulation boundary after the optimizer/scaler/scheduler transition completes without exception. Any exception in that transition prevents a success metric and leaves the Trainer fail-closed (`_update_incomplete` / `_failure_reason`).
- Ambiguous or partially committed optimizer/scheduler failure cannot silently continue, serialize, or repair the same Trainer in place; recovery requires a fresh Trainer around a verified checkpoint.
- `optimizer_step` increments exactly once inside the successful transition; partial accumulation microbatches return `optimizer_stepped=False` and do not increment it.
- `update_loss` is the token-weighted mean over the full accumulation window. `_count_training_tokens` matches the actual causal-loss denominator for both shifted raw labels and aligned `target_ids + loss_mask`.
- Finite-step tests prove parameters actually change when `optimizer_stepped=True`; variable-token accumulation matches one combined batch; warmup telemetry reports the LR applied to that update.
- Checkpoint hooks run only after committed updates; checkpoint-hook failure is explicitly wrapped as `CheckpointHookError` saying the committed step must not be blindly replayed.
- Checkpoint-safe serialization enforces complete accumulation boundary, exact `micro_step == optimizer_step * gradient_accumulation_steps`, zero pending statistics, and non-poisoned state. Trainer-state restore is transactional and rolls back partial component mutation on failure.
- Exact-head CI run `34536831375` succeeded on the target SHA: preflight, locked-arm64, locked-x86-64 green; locked x86 full checks reported `1223 passed`.

Adversarial note:
- `on_metrics` callback exceptions are not wrapped in a dedicated committed-step exception. This is an observability/API nuance, not a demonstrated false-commit path: the callback is invoked only after `train_microbatch` has returned the already-advanced truthful trainer state. No Product change is required by this audit.

TRUTH BOUNDARY:
- This PASS is narrow. It does NOT promote or merge PR #802.
- It does NOT authorize the bounded pilot #1248 or any optimized-target exposure.
- It does NOT resolve #1264's D05 checkpoint-selection/recovery finding, D04 exposure-guard requirements, corpus gates, recipe authority, or evaluation authority.
- No training was launched; no learned weights were created; final-test outcomes were not read; no paid compute/foreign pretrained weights/external LLM corpus construction were used.

SWARM-1300 releases `AUDIT-B|D02|PR802|TRAIN-MICROBATCH-COMMIT-METRIC-ACCOUNTING|EXACTHEAD-82C43005|V1` as DONE_PASS.
